src/VideoToolbox/VTVideoEncoder.cs

```diff
@@ -49,82 +51,111 @@ public class VTVideoEncoder {
 		public string EncoderName { get; private set; }
 
 #if NET
+		[SupportedOSPlatform ("macos10.14.6")]
 		[SupportedOSPlatform ("ios13.0")]
 		[SupportedOSPlatform ("tvos13.0")]
-		[SupportedOSPlatform ("macos10.14.6")]
 #else
-		[Mac (10,14,6), iOS (13,0), TV (13,0)]
+		[Mac (10,14,6)]
+		[iOS (13,0)]
+		[TV (13,0)]
 #endif
 		public ulong? GpuRegistryId { get; private set; } // optional, same type as `[MTLDevice registryID]`
 
 #if NET
+		[SupportedOSPlatform ("macos10.14.6")]
 		[SupportedOSPlatform ("ios13.0")]
 		[SupportedOSPlatform ("tvos13.0")]
-		[SupportedOSPlatform ("macos10.14.6")]
 #else
-		[Mac (10,14,6), iOS (13,0), TV (13,0)]
+		[Mac (10,14,6)]
+		[iOS (13,0)]
+		[TV (13,0)]
 #endif
 		public NSDictionary SupportedSelectionProperties {get; private set; }
 
 #if NET
+		[SupportedOSPlatform ("macos10.14.6")]
 		[SupportedOSPlatform ("ios13.0")]
 		[SupportedOSPlatform ("tvos13.0")]
-		[SupportedOSPlatform ("macos10.14.6")]
 #else
-		[Mac (10,14,6), iOS (13,0), TV (13,0)]
+		[Mac (10,14,6)]
+		[iOS (13,0)]
+		[TV (13,0)]
 #endif
 		public NSNumber PerformanceRating { get; private set; }
 
 #if NET
+		[SupportedOSPlatform ("macos10.14.6")]
 		[SupportedOSPlatform ("ios13.0")]
 		[SupportedOSPlatform ("tvos13.0")]
-		[SupportedOSPlatform ("macos10.14.6")]
 #else
-		[Mac (10,14,6), iOS (13,0), TV (13,0)]
+		[Mac (10,14,6)]
+		[iOS (13,0)]
+		[TV (13,0)]
 #endif
 		public NSNumber QualityRating { get; private set; }
 
 #if NET
+		[SupportedOSPlatform ("macos10.14.6")]
 		[SupportedOSPlatform ("ios13.0")]
 		[SupportedOSPlatform ("tvos13.0")]
-		[SupportedOSPlatform ("macos10.14.6")]
 #else
-		[Mac (10,14,6), iOS (13,0), TV (13,0)]
+		[Mac (10,14,6)]
+		[iOS (13,0)]
+		[TV (13,0)]
 #endif
 		public bool? InstanceLimit { get; private set; }
 
 #if NET
+		[SupportedOSPlatform ("macos10.14.6")]
 		[SupportedOSPlatform ("ios13.0")]
 		[SupportedOSPlatform ("tvos13.0")]
-		[SupportedOSPlatform ("macos10.14.6")]
 #else
-		[Mac (10,14,6), iOS (13,0), TV (13,0)]
+		[Mac (10,14,6)]
+		[iOS (13,0)]
+		[TV (13,0)]
 #endif
 		public bool? IsHardwareAccelerated { get; private set; }
 
 #if NET
 		[SupportedOSPlatform ("ios14.2")]
 		[SupportedOSPlatform ("tvos14.2")]
-		[SupportedOSPlatform ("maccatalyst14.2")]
 		[SupportedOSPlatform ("macos11.0")]
+		[SupportedOSPlatform ("maccatalyst14.2")]
 #else
-		[iOS (14,2)][TV (14,2)][Mac (11,0)]
+		[iOS (14,2)]
+		[TV (14,2)]
+		[Mac (11,0)]
 		[MacCatalyst (14,2)]
 #endif
 		public bool SupportsFrameReordering { get; private set; }
 
 #if !NET
-		[NoiOS, NoTV, NoMacCatalyst, NoMac, NoWatch]
+#if NET
+		[UnsupportedOSPlatform ("ios")]
+		[UnsupportedOSPlatform ("tvos")]
+		[UnsupportedOSPlatform ("maccatalyst")]
+		[UnsupportedOSPlatform ("macos")]
+#else
+		[NoiOS]
+		[NoTV]
+		[NoMacCatalyst]
+		[NoMac]
+		[NoWatch]
+#endif
 		public bool SupportsMultiPass { get; private set; }
 #endif // !NET
```

Comment:
This concerns me. How big is this !NET block? If it includes much this file so far, you can revert all attributes inside since we'll never have them in NET.